core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1178,6 +1178,37 @@ abstract class RDD[T: ClassTag](
     }
   }
 
+  /**
+   * Computing desired tree aggregate depth necessary to avoid exceeding
+   * driver.MaxResultSize during aggregation.
+   * Based on the formulae: (numPartitions)^(1/depth) * objectSize <= DriverMaxResultSize
+   * @param aggregatedObjectSizeInMb the size, in megabytes, of the object being tree aggregated
+   * @param numPartitions the number of partitions for which to aggregate partial results
+   * @param maxDriverResultSizeInMb the value of the parameter spark.driver.maxResultSize in MB
+   */
+  def getTreeAggregateIdealDepth(
```

Comment:
This shouldn't be in RDD; it can just live in RowMatrix for now. it can be `private[spark]` for testing.

Reply:
Fine, I'll move that (and the test to RowMatrix class)